gating/pre_merge_test/run_deploy_mnaio.sh

```diff
@@ -0,0 +1,144 @@
+export OSA_PORTS="6080 6082 443 80 8443"
+export RPC_BRANCH="${RE_JOB_BRANCH}"
+export DEFAULT_MIRROR_HOSTNAME=mirror.rackspace.com
+export DEFAULT_MIRROR_DIR=/ubuntu
+export INFRA_VM_SERVER_RAM=16384
+
+# ssh command used to execute tests on infra1
+export MNAIO_SSH="ssh -ttt -oStrictHostKeyChecking=no root@infra1"
+
+## Functions -----------------------------------------------------------------
+source /opt/rpc-openstack/scripts/functions.sh
+
+## Main --------------------------------------------------------------------
+
+# capture all RE_ variables
+env | grep RE_ | sed 's/^/export /' > /opt/rpc-openstack/RE_ENV
+
+# checkout openstack-ansible-ops
+if [ ! -d "/opt/openstack-ansible-ops" ]; then
+  git clone --recursive https://github.com/openstack/openstack-ansible-ops /opt/openstack-ansible-ops
+else
+  pushd /opt/openstack-ansible-ops
+    git fetch --all
+  popd
+fi
+
+# apply various modifications for mnaio
+pushd /opt/openstack-ansible-ops/multi-node-aio
+  # By default the MNAIO deploys metering services, so we override
+  # osa_enable_meter to prevent those services from being deployed.
+  sed -i 's/osa_enable_meter: true/osa_enable_meter: false/' playbooks/group_vars/all.yml
+popd
+
+# build the multi node aio
+pushd /opt/openstack-ansible-ops/multi-node-aio
+  ./build.sh
+popd
+echo "Multi Node AIO setup completed..."
+
+# capture all RE_ variables for push to infra1
+env | grep RE_ | sed 's/^/export /' > /opt/rpc-openstack/RE_ENV
+
+# check if we're using artifacts or not
+if [[ ${RE_JOB_IMAGE} =~ no_artifacts$ ]]; then
+  echo "export RPC_APT_ARTIFACT_ENABLED=no" >> /opt/rpc-openstack/RE_ENV
+  ${MNAIO_SSH} "apt-get -qq update; DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade"
+elif [[ ${RE_JOB_IMAGE} =~ loose_artifacts$ ]]; then
+  # Set the apt artifact mode
+  echo "export RPC_APT_ARTIFACT_MODE=loose >> /opt/rpc-openstack/RE_ENV
+  ${MNAIO_SSH} "apt-get -qq update; DEBIAN_FRONTEND=noninteractive apt-get -y dist-upgrade"
+fi
+
+# Set the appropriate scenario variables
+if [[ "${RE_JOB_SCENARIO}" == "ceph" ]]; then
+  echo "export DEPLOY_CEPH=yes >> /opt/rpc-openstack/RE_ENV
+  echo "export DEPLOY_SWIFT=no" >> /opt/rpc-openstack/RE_ENV
+elif [[ "${RE_JOB_SCENARIO}" == "ironic" ]]; then
+  echo "export DEPLOY_IRONIC=yes" >> /opt/rpc-openstack/RE_ENV
+fi
+
+# prepare rpc-o configs
+set -xe
+scp -r -o StrictHostKeyChecking=no /opt/rpc-openstack infra1:/opt/
+ssh -T -o StrictHostKeyChecking=no infra1 << 'EOF'
+set -xe
+sudo cp /etc/openstack_deploy/user_variables.yml /etc/openstack_deploy/user_variables.yml.bak
+sudo cp -R /opt/openstack-ansible/etc/openstack_deploy /etc
+sudo cp /etc/openstack_deploy/user_variables.yml.bak /etc/openstack_deploy/user_variables.yml
+sudo cp -R /opt/rpc-openstack/etc/openstack_deploy/* /etc/openstack_deploy/
+sudo rm -rf /opt/openstack-ansible
+sudo rm /usr/local/bin/openstack-ansible
+EOF
+
+# start the rpc-o install from infra1
+${MNAIO_SSH} "source /opt/rpc-openstack/RE_ENV; \
```

Comment:
I'd change this to use a heredoc. and add the line to ensure we're testing with artifacts, even if they're being disabled. 

``` shell
${MNAIO_SSH} <<EOC
  source /opt/rpc-openstack/RE_ENV
  pushd /opt/rpc-openstack
    scripts/deploy.sh
  popd
  pushd /opt/rpc-openstack/playbooks
    openstack-ansible -i 'localhost,' \
                      -e 'apt_target_group=localhost' \
                      -e "apt_artifact_enabled='${RPC_APT_ARTIFACT_ENABLED}'" \
                      -e "apt_artifact_mode='${RPC_APT_ARTIFACT_MODE}'" \
                      -e "container_artifact_enabled='${RPC_CONTAINER_ARTIFACT_ENABLED}'" \
                      site-artifacts.yml
    openstack-ansible openstack-ansible-install.yml
  popd
  pushd /opt/openstack-ansible/scripts
    python pw-token-gen.py --file /etc/openstack_deploy/user_secrets.yml
    pushd /opt/openstack-ansible/playbooks
      openstack-ansible setup-everything.yml
    popd
  popd
EOC
```

Reply:
Note the environment variables in the previous comment would need to be added too.

RPC_CONTAINER_ARTIFACT_ENABLED
RPC_APT_ARTIFACT_ENABLED
RPC_APT_ARTIFACT_MODE